src/sentry/issues/data_export.py

```diff
@@ -0,0 +1,203 @@
+
+    while True:
+        # Get data from queue (blocks until available)
+        data = upload_queue.get()
+
+        if data is None:  # Stop signal
+            break
+
+        file_counter += 1
+        upload_to_gcs(data, file_counter, bucket, gcs_prefix)
+
+
+def add_events_to_upload_queue(events_data: list[dict], upload_queue: Queue):
+    jsonl_data = []
+    for event_data in events_data:
+        if event_data:
+            jsonl_data.append(json.dumps(event_data))
+
+    if not jsonl_data:
+        return
+
+    jsonl_content = "\n".join(jsonl_data) + "\n"
+    compressed_data = gzip.compress(jsonl_content.encode("utf-8"))
+
+    upload_queue.put(compressed_data)
+
+
+def process_event_batch(events: list[Event], organization_id: int) -> list[dict]:
+    results = []
+
+    for event in events:
+        if event.data:
+            # TODO: some of this metadata might not be necessary
+            event_data = {
+                "event_id": event.event_id,
+                "project_id": event.project_id,
+                "organization_id": organization_id,
+                "timestamp": event.datetime.isoformat(),
+                "platform": event.platform,
+                "message": event.message,
+                "type": event.get_event_type(),
+                "data": dict(event.data),  # Full event payload
+            }
+            results.append(event_data)
+        else:
+            logger.warning("No data for event", extra={"event_id": event.event_id})
+
+    return results
+
+
+def process_event_batches(
+    event_batches: Iterator[list[Event]],
+    destination_bucket: str,
+    gcs_prefix: str,
+    organization_id: int,
+):
+    # Create upload queue and background uploader
+    upload_queue = Queue(maxsize=UPLOAD_QUEUE_SIZE)
+    upload_thread = threading.Thread(
+        target=background_uploader,
+        args=(upload_queue, destination_bucket, gcs_prefix),
+        daemon=True,
+    )
+    upload_thread.start()
+
+    for batch in event_batches:
+        events_data = process_event_batch(batch, organization_id)
+        add_events_to_upload_queue(events_data, upload_queue)
+
+    upload_queue.put(None)  # Signal upload thread to finish
+    upload_thread.join(timeout=300)  # Wait for upload thread to complete - 5 minute timeout
+
+
+def get_event_batches(
+    organization_id: int, event_filter: eventstore.Filter
+) -> Iterator[list[Event]]:
+    offset = 0
+    while True:
+        event_batch = eventstore.backend.get_events(
+            filter=event_filter,
+            limit=BATCH_SIZE,
+            offset=offset,
+            referrer=REFERRER,
+            tenant_ids={"organization_id": organization_id},
+        )
+
+        if not event_batch:
+            break
+
+        yield event_batch
+        offset += BATCH_SIZE
+
+
+# TODO: run as task?
+@instrumented_task(
+    name="sentry.issues.tasks.export_project_errors_async",
+    taskworker_config=TaskworkerConfig(namespace=issues_tasks),
+)
```

Comment:
My understanding is that this will be invoked by workflow engine via command line. There will be a parent script that invokes your code as a function, see https://www.notion.so/sentry/EU-Data-Export-Design-Document-26b8b10e4b5d80fba82ff28771a3f82b